### PR TITLE
estimation-no-signature

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- Estimation without signature

## Why

- UX improvement to not require additional signed tx

## Refs

- https://github.com/rsksmart/rif-relay-contracts/pull/168
- It solves #92

### CODE REVIEW

**Comment by franciscotobar** [comment]
We still need to decide the `version`.

File: package.json [hunk 1]
```diff
@@ -90,7 +90,7 @@
     "typescript": "4.8.2"
   },
   "peerDependencies": {
-    "@rsksmart/rif-relay-contracts": "^2.1.1-beta.0",
+    "@rsksmart/rif-relay-contracts": "github:rsksmart/rif-relay-contracts#estimation-no-signature",
```
Comment (antomor) [inline package.json:93]: To be changed once we merge the contracts

File: src/utils.ts [hunk 2]
```diff
@@ -121,7 +123,7 @@ const estimatePaymentGas = async ({
     // This may be wrong in case the feesReceiver performs some operations using
     // the msg.sender address in the fallback/receive function.
     return await estimateInternalCallGas({
-      from: feesReceiver,
+      from: tokenOrigin,
```
Comment (antomor) [inline src/utils.ts:126]: If we introduce this back, we'll need to change the comments to clarify that it works on rskj, while in other EVMs may not work as expected (Anvil is an example)

**Review by antomor — APPROVED** [review]
LGTM

**Review by antomor — APPROVED** [review]
LGTM

File: src/utils.ts (hunk 2 above)
Comment (antomor) [reply src/utils.ts:126]: I decided to include a check against the chainID and set it to `tokenOrigin` in rskj and to `feesReceiver` in other cases

**Comment by antomor** [comment]
We decided to abandon the linear fit implementation and use a different approach, taking into account the costs involved into the different stages of the transaction and including some cushions to be sure the estimation is enough for executing the transaction.

**Pull request merged**